opentelemetry-sdk/src/opentelemetry/sdk/trace/export/experimental/timer.py

```diff
@@ -0,0 +1,111 @@
+import abc
+import threading
+import typing
+
+
+class TimerABC(abc.ABC):
+    """
+    An interface extracted from PeriodicTimer so alternative implementations can be used for testing.
+    """
+
+    @abc.abstractmethod
+    def set_callback(self, cb) -> None:
+        pass
+
+    @abc.abstractmethod
+    def start(self) -> None:
+        pass
+
+    @abc.abstractmethod
+    def poke(self) -> None:
+        pass
+
+    @abc.abstractmethod
+    def stop(self) -> None:
+        pass
+
+    @abc.abstractmethod
+    def started(self) -> None:
+        pass
+
+    @abc.abstractmethod
+    def stopped(self) -> None:
+        pass
+
+
+class PeriodicTimer(TimerABC):
+    """
+    Executes the passed-in callback on a timer at the specified interval. The callback can be run sooner than the
+    interval via the poke() method, which also resets the timer.
+    """
+
+    def __init__(
+        self,
+        interval_sec: int,
+        callback: typing.Callable[[], None] = lambda: None,
+        daemon: bool = True,
```

Comment:
Consider adding `self._lock = threading.Lock()` to ensure thread safety.

Reply:
Thanks Nina for the feedback -- I have attempted to address the locking suggestions below. Will also add some comments to the code to indicate what's thread safe and what's not.